### PR TITLE
Improve GitHub Actions

### PR DESCRIPTION
**Pull request by natemoo-re**
## Changes

- Updates and consolidates our CI process on GitHub Actions.
- Previously, our CI ran across many separate actions, make it hard to inspect at a glance.
- With these changes, CI will run in a single job with dependent steps.
- The full `yarn install` and `yarn build` only run once, so dependent jobs should be as fast as possible. Hard to track the exact speed boost across our current jobs, but it seems significantly faster in my testing.

Screen Shot 2021-11-29 at 9 37 36 AM

## Testing

N/A, QoL improvement only

## Docs

N/A, QoL improvement only

### CODE REVIEW

File: .github/workflows/congratsbot.yml [hunk 1]
```diff
@@ -1,24 +0,0 @@
-name: "Discord:congratsbot"
```
Comment (natemoo-re) [inline .github/workflows/congratsbot.yml:1]: This job was moved to `main.yml`

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -1,53 +0,0 @@
-name: Changelog
```
Comment (natemoo-re) [inline .github/workflows/release.yml:1]: This job was moved to `ci.yml` to run after tests pass

File: .github/workflows/stat.yml [hunk 3]
```diff
@@ -7,7 +7,7 @@ on:
 
 jobs:
   stat:
-    if: github.repository == 'withastro/astro'
+    if: github.repository_owner == 'withastro'
```
Comment (natemoo-re) [inline .github/workflows/stat.yml:10]: Very small change, just a little cleaner

File: scripts/notify/index.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+  if (packages.length === 1) {
+    const { name, version, url } = packages[0];
+    console.log(`\`${name}@${version}\` was just released! Read the [release notes →](<${url}>)`);
+  } else {
+    console.log(`**Some new releases just went out!**\n`);
+    for (const { name, version, url } of packages) {
+      console.log(`  • \`${name}@${version}\` ([Release Notes →](<${url}>))`);
+    }
+  }
+}
```
Comment (natemoo-re) [inline scripts/notify/index.js:23]: Our release notes can now specify the `package@version` that has been published and point directly to that version's changelog.

File: .changeset/config.json [hunk 5]
```diff
@@ -1,6 +1,6 @@
 {
   "$schema": "https://unpkg.com/@changesets/config@1.6.0/schema.json",
-  "changelog": "@changesets/cli/changelog",
+  "changelog": "@changesets/changelog-github",
```
Comment (natemoo-re) [inline .changeset/config.json:3]: This updates our CHANGELOG generator to include links to the commit SHA, PR links, and a thank you to the author.

File: .github/workflows/main.yml [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+    branches:
+      - main
+
+env:
+  GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+# Automatically cancel in-progress actions on the same branch
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event_name == 'pull_request' && github.head_ref || github.ref }}
+  cancel-in-progress: true
```
Comment (drwpow) [inline .github/workflows/main.yml:14]: Nice!

**Review by drwpow — APPROVED** [review]
Great improvement! LGTM

**Comment by matthewp** [comment]
lgtm too

**Comment by FredKSchott** [comment]
lol love the commit history :)

This is a total nit comment, especially since I missed the original review, but thought I'd share for the historic record: Personally I don't like a CI task that pushes changes to my branch for me, invisibly. it's easy to miss that I need to synchronize my branch with origin, and I'll often end up with nasty rebase/merge conflicts trying to merge some new work onto the latest PR branch on origin.

to be fair this comes more from `format` experience than `lint`, where auto formatting changes are generally more noisy. 

Again, all personal preference! Also, I may be misunderstanding what "autofix" does, in which case ignore me :) Either way, I'm happy to try this out for a bit and if the rest of the team likes it I'm not against it.

**Pull request merged**